drivers/SmartThings/zwave-switch/src/qubino-switches/qubino-relays/qubino-flush-2-relay/init.lua

```diff
@@ -11,43 +11,163 @@
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
+local st_device = require "st.device"
+local capabilities = require "st.capabilities"
+--- @type st.zwave.CommandClass
+local cc = require "st.zwave.CommandClass"
+--- @type st.zwave.CommandClass.Basic
+local Basic = (require "st.zwave.CommandClass.Basic")({ version = 1 })
+--- @type st.zwave.CommandClass.SwitchBinary
+local SwitchBinary = (require "st.zwave.CommandClass.SwitchBinary")({ version = 1 })
+--- @type st.zwave.CommandClass.Meter
+local Meter = (require "st.zwave.CommandClass.Meter")({ version = 3 })
+--- @type st.zwave.CommandClass.SensorMultilevel
+local SensorMultilevel = (require "st.zwave.CommandClass.SensorMultilevel")({ version = 7 })
+--- @type st.zwave.constants
+local constants = require "st.zwave.constants"
 
-local QUBINO_FLUSH_2_RELAY_FINGERPRINT = {mfr = 0x0159, prod = 0x0002, model = 0x0051}
+local QUBINO_FLUSH_2_RELAY_FINGERPRINT = { mfr = 0x0159, prod = 0x0002, model = 0x0051 }
 
 local function can_handle_qubino_flush_2_relay(opts, driver, device, ...)
   return device:id_match(QUBINO_FLUSH_2_RELAY_FINGERPRINT.mfr, QUBINO_FLUSH_2_RELAY_FINGERPRINT.prod, QUBINO_FLUSH_2_RELAY_FINGERPRINT.model)
 end
 
 local function component_to_endpoint(device, component_id)
-    if component_id == "main" then
-      return { 1 }
-    elseif component_id == "extraTemperatureSensor" then
-      return { 3 }
-    else
-      local ep_num = math.floor(component_id:match("switch(%d)"))
-      return { ep_num and tonumber(ep_num)}
+  return { 1 }
+end
+
+local function find_child(parent, src_channel)
+  if src_channel == 1 then
+    return parent
+  else
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", src_channel))
+  end
+end
+
+local function device_init(driver, device)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    device:set_find_child(find_child)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
+  end
+end
+
+local function get_child_metadata(device, endpoint)
+  local name
+  local profile_name
+  if endpoint ~= 3 then
+    name = string.format("Qubino Switch %d", endpoint)
+    profile_name = "metering-switch"
+  else
+    name = "Qubino Temperature Sensor"
+    profile_name = "child-temperature"
+  end
+
+  return {
+    type = "EDGE_CHILD",
+    label = name,
+    profile = profile_name,
+    parent_device_id = device.id,
+    parent_assigned_child_key = string.format("%02X", endpoint),
+    vendor_provided_label = name
+  }
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD then
+    driver:try_create_device(get_child_metadata(device, 2))
+    if device:is_cc_supported(cc.SENSOR_MULTILEVEL) then
+      driver:try_create_device(get_child_metadata(device, 3))
     end
+  end
+end
+
+local function send_refresh_to_endpoint(ep, driver, device)
```

Comment:
Instead of having this function take an endpoint and depend on that to determine what refresh commands should be sent, can we not just use the device profile/capabilities supported to determine it?

Reply:
Here is an example of what I'm suggesting: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/commit/0f1e5807ecde7a82285405ba496e3719bbba8946

Reply:
Done